### ISSUE TITLE
what are replaced methods for cachedImageExistsForURL: and diskImageExistsForURL: in version 4.0?

### ISSUE THREAD

**Issue by singhjagjot1992** [rs/SDWebImage#1920]
### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [ ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / tvos
 Platform Version        | e.g. 8.0
 SDWebImage Version      | e.g. 3.7.6
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 7.3
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**Comment by bpoplauschi** [comment]
@singhjagjot1992 I might be missing something, but this is a question you can answer very quickly by just looking at the code or reading the docs (the removal of those methods was added to the Changelog and the release notes section).

Look at `cachedImageExistsForURL:completion:` and `diskImageExistsForURL:completion:`

**Comment by cdunkel** [comment]
@bpoplauschi, I know this issue was closed long ago, but I'm just migrating to 4.3.3 and I'm looking for best practices for checking if an image is in the cache synchronously. I have many places in my code where I just want to verify if an image has been downloaded, but modifying the code to do that asynchronously will require a large amount of re-architecting.

So far the only alternative I can see is using `imageFromDiskCacheForKey:` to actually load the whole image into memory and see if that was successful. That seems very inefficient.

Thank you for any guidance you can provide!

**Comment by bpoplauschi** [comment]
I suggest `diskImageDataExistsWithKey:` (the sync one)

**Comment by cdunkel** [comment]
Thanks! I didn't see that in the docs.

**Comment by cdunkel** [comment]
Hi @bpoplauschi or @dreampiggy. I'm back! I'm now upgrading SDWebImage to v5.0.5 and it looks like `diskImageDataExistsWithKey:` has either been removed or has been made private. Is that the case? If so, is there another way to synchronously check if an image has been cached?

I'm trying, wherever possible, to use the new asynchronous methods, but I have one situation that is very time-sensitive. I'm showing an image to the user on a splash screen while content loads behind it and I don't have time to wait for the image to download if it hasn't already been cached...I have a different view that I will show to them of no images are available.

Any guidance on the best way to approach that situation?

Thanks!

**Comment by dreampiggy** [comment]
Why not just using the `SDImageCacheProtocol` method ? That do what you need.
See documentation: https://sdwebimage.github.io/Protocols/SDImageCache.html#/c:objc(pl)SDImageCache(im)containsImageForKey:cacheType:completion:

**Comment by dreampiggy** [comment]
And, the `diskImageDataExistsWithKey:` is not private. I assume you have something install issue or header include issue.
See: https://sdwebimage.github.io/Classes/SDImageCache.html#/c:objc(cs)SDImageCache(im)diskImageDataExistsWithKey:

**Comment by dreampiggy** [comment]
This return type change is the main feature of 5.0. Now, the manager does not knwo what cache class is. You can custom you own cache other than `SDImageCache` class, like `YYCache`, `MMKV`, any of them.

See release log: https://github.com/SDWebImage/SDWebImage/releases/tag/5.0.0
Wiki: https://github.com/SDWebImage/SDWebImage/wiki/Advanced-Usage#custom-cache-50

For your case, if you don't use any of custom cache implementation, you can just using `SDImageCache.shared` to call the method, it's the same.

**Comment by cdunkel** [comment]
@dreampiggy, thanks for the great answers!

Using `containsImageForKey:cacheType:completion:` would have required re-architechting some portions of my app to allow for an asynchronous check. Accessing the cache using `SDImageCache.shared` ended up working just fine.

Thanks again!
